presto-rcfile/src/main/java/com/facebook/presto/rcfile/RcFileReader.java

```diff
@@ -286,11 +287,11 @@ public void close()
         rowGroupPosition = 0;
         rowGroupRowCount = 0;
         currentChunkRowCount = 0;
-        input.close();
-        if (decompressor != null) {
-            decompressor.destroy();
-        }
 
+        try (Closer closer = Closer.create()) {
+            closer.register(input);
+            closer.register(decompressor::destroy);
+        }
```

Comment:
Lets do this with a traditional `try/finally` as it is more obvious what is going on.  If there were more things going on in this code (e.g., loops, conditional code, etc), then closer might make sense:
```java
try  {
    input.close();
} finally {
    if (decompressor != null) {
        decompressor.destroy();
    }
}